substrate/frame/balances/src/impl_fungible.rs

```diff
@@ -356,3 +359,16 @@ impl<T: Config<I>, I: 'static> fungible::Balanced<T::AccountId> for Pallet<T, I>
 }
 
 impl<T: Config<I>, I: 'static> fungible::BalancedHold<T::AccountId> for Pallet<T, I> {}
+
+impl<T: Config<I>, I: 'static> AccountTouch<(), T::AccountId> for Pallet<T, I> {
+	type Balance = T::Balance;
+	fn deposit_required(_: ()) -> Self::Balance {
+		Self::Balance::zero()
+	}
+	fn should_touch(_: (), _: &T::AccountId) -> bool {
+		false
```

Comment:
this is because there is a guarantee that if you transfer some amount which is greater than ED, the account will be created. it wont fail because of some account creation requirements. 
initially this trait was introduced for the asset conversion pallet, so that the pool account creation for some asset can be guaranteed. otherwise someone for example can derive future pool account ids and exhaust all consumer references.
I also tried to explain this in docs for `AccountTouch` trait.